### PR TITLE
podman 3.2.3

### PR DESCRIPTION
**Pull request by ashley-cui**
Add gvproxy resource for podman networking

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by ashley-cui** [comment]
adding as a resource as per https://github.com/Homebrew/homebrew-core/pull/83449#issuecomment-901067238

**Comment by SMillerDev** [comment]
Retrying this with a linux build.

**Comment by ashley-cui** [comment]
@SMillerDev  This formula should be darwin-only

**Comment by carlocab** [comment]
Linux probably does not like `make podman-remote-darwin` or `make install-podman-remote-darwin-docs`.

**Comment by carlocab** [comment]
Are there targets in your `Makefile` for Linux?

**Comment by SMillerDev** [comment]
Well... no... but podman is hardly a macOS specific tool. If anything it's more Linux specific.

**Comment by SMillerDev** [comment]
https://github.com/containers/podman/blob/main/Makefile#L331 and https://github.com/containers/podman/blob/main/Makefile#L446

**Comment by ashley-cui** [comment]
I'd prefer we don't allow linux builds, since this would be called podman-remote on linux, but podman on mac. But if we do, can we name the binary on linux podman-remote, and on mac podman? This makes things super complicated for us since we also maintain linux builds in specific linux package managers.

**Comment by SMillerDev** [comment]
> I'd strongly prefer we don't allow linux builds. This makes things super complicated for us since we also maintain linux builds in specific linux package managers. 

Homebrew has been available on macOS and Linux for a while and we're currently taking the final steps to merge the Linux and macOS work. In addition to that, we don't really have the concept of package maintainers, so while I appreciate the work you're putting into homebrew it's your package as much as it is anyone's.

> Podman on linux is the full podman, but on mac it's actually a remote client. But if we do, can we name the binary on linux podman-remote, and on mac podman?

I'd rather name it podman-remote on macOS too then, if it isn't a "real" podman install.

**Comment by ashley-cui** [comment]
> I'd rather name it podman-remote on macOS too then, if it isn't a "real" podman install.
We cannot do that either, since we're trying to make it as close to the native linux experience as possible on mac. Naming it podman-remote on mac would not do well for this.

**Comment by SMillerDev** [comment]
> We cannot do that either, since we're trying to make it as close to the native linux experience as possible on mac. Naming it podman-remote on mac would not do well for this.

But the native linux experience would name the same thing "podman-remote". It's very confusing that you have `A` and `B`, where only `B` works on macOS. But on macOS you're calling `B` by the name `A` to make it closer to the experience on Linux, where you have `A` and not `B`.

**Comment by ashley-cui** [comment]
The point is that we're abstracting the VM that podman is actually running in. When podman is run on the mac, we're connecting to a vm (like in docker) behind the scenes, so we're using the remote client. So, basically the command syntax on macs and linux would be the same if they're both named `podman`, just the backend execution would be different. This is why we decided to name it podman on a mac.

**Comment by ashley-cui** [comment]
Renaming the binary would also break a lot of users right now.

**Comment by SMillerDev** [comment]
Isn't it just much easier for everyone if `podman-remote` is podman, but remotely and people don't mistakenly think that `podman` on macOS works/exists?

> Renaming the binary would also break a lot of users right now.

That's true, but we can make a symlink so people can keep using the old alias.

CC @Homebrew/maintainers for opinions?

**Comment by ashley-cui** [comment]
@SMillerDev , the idea is that `podman` on macOS does work, it just works on a vm. It would feel exactly like podman on a linux, where calling `podman` does everything that `podman` does on linux.

**Comment by danielnachun** [comment]
>This makes things super complicated for us since we also maintain linux builds in specific linux package managers.
We have maintainers with considerable experience building things on Linux (one of whom is me), so upstream developers need not worry about maintaining the Linux builds on their own.

We've had a lot of interest in providing the full `podman` to Linux, so I'd like us to get it built there.  We've added what we think are all the necessary Linux-only dependencies to do this, and can make the extra steps to build the full `podman` Linux-only.  As @SMillerDev has suggested, we can simply create a macOS-only symlink from `podman-remote` to `podman`, so that this change is seamless for macOS users, while for Linux users, `podman` and `podman-remote` remain separate.  

With all of that in mind, I'm okay for this version bump if we just get the `podman-remote` target to build on Linux.  We can make a separate PR to build the full thing on Linux.

**Comment by ashley-cui** [comment]
I need to check with the podman team as well, but the expectation would be on macs `brew install podman` would allow people to use podman-remote via the `podman` command. As long as the symlink from `podman` to podman-remote always exists on macs, I think that would be fine. But expecting `podman` to be renamed to `podman-remote` on macs would not be fine. 

cc @baude, @rhatdan

**Comment by baude** [comment]
I would be willing to support the notion of the symlink as well.  Like @ashley-cui, I would not support a forced rename on MacOS.  If you are having build problems on Linux, several of us may be able to help.  We hang out on #podman on libera.

File: Formula/podman.rb
Comment (danielnachun) [inline Formula/podman.rb:53]: I guess you need `assert_match "podman-remote version #{version}"` for the test to work now.

**Comment by danielnachun** [comment]
It sounds like we are in agreeement then that we can just make a symlink on macOS where `podman` points to `podman-remote`, while on Linux, it will eventually point to the full `podman`.  As I said above we can open a follow up PR to build the rest of `podman` on Linux.  I'll have to test locally in Docker CI first because we have to add some dependencies and make other changes.  I'll reach out if I hit any weird problems with the full Linux build.

**Comment by ashley-cui** [comment]
Sounds good, thanks for all your help! I can fix up this PR :) @danielnachun @SMillerDev

File: Formula/podman.rb
Comment (MikeMcQuaid) [inline Formula/podman.rb:29]:
```suggestion
    os = if OS.mac?
      "darwin"
    else
      "linux"
    end
```

apologies if this contradicts what others have said but @Homebrew/maintainers I think this is an example of a case where the if/else form is dramatically easier to read. Not sure if an audit/style check forbids this but: if it does, we should make it not apply for `def install` (and not let that block this PR).

File: Formula/podman.rb
Comment (MikeMcQuaid) [inline Formula/podman.rb:37]: Not blocking this PR but for a future release it would be nice to have an upstream `make install` which encapsulates this logic.

File: Formula/podman.rb
Comment (carlocab) [reply Formula/podman.rb:29]: This pattern actually appears all over the place. It's part of what motivated me to open Homebrew/brew#11687 but neglected to include it when I was writing things down 😅

Comment (MikeMcQuaid) [reply Formula/podman.rb:29]: Yeh, ouch. Would be good to improve this but definitely not blocking this PR then.

**Review by SMillerDev — APPROVED** [review]

File: Formula/podman.rb
Comment (dawidd6) [reply Formula/podman.rb]: Mixing those forms brings some confusion and lack of consistency. I don't think we should be using both interchangeably, the readability is in my opinion on par in both cases. If we start using `OS.mac?` again here and there, then what was the point in the first place changing all those occurrences earlier. I would say we stay with `on_os` blocks exclusively.

Comment (carlocab) [reply Formula/podman.rb]: I think the `if` form is significantly better:
1. it's idiomatic Ruby, which means one doesn't need to be a regular Homebrew/core contributor to understand at a glance what's going on
2. `on_os` blocks also have slightly more complicated variable scoping rules

Comment (Bo98) [reply Formula/podman.rb]: This one is so common we could even have a `OS.uname`/`kernel_name`.

Comment (dawidd6) [reply Formula/podman.rb]: If so then I would say drop `on_os` blocks altogether and just use `if` statements like before and everywhere else. Then the revisions problem would automatically go away cause we could just migrate those too and that's it.

Comment (carlocab) [reply Formula/podman.rb]: See Homebrew/brew#11687 for a discussion of why we still want `on_os` blocks even if we're not using them in `install` methods.

Comment (MikeMcQuaid) [reply Formula/podman.rb]: > If so then I would say drop `on_os` blocks altogether and just use `if` statements like before and everywhere else.

I'd be fine with doing that in `install`/`post_install`/`test` methods but definitely not in the class definition.

**Pull request closed without merging**
